packages/backend/server/src/__tests__/utils/testing-app.ts

```diff
@@ -156,12 +156,18 @@ export class TestingApp extends ApplyType<INestApplication>() {
       .send({
         query,
         variables,
-      })
-      .expect(200);
+      });
 
-    if (res.body.errors?.length) {
+    if (res.body?.errors?.length) {
       throw new Error(res.body.errors[0].message);
     }
+    if (res.status !== 200) {
```

Comment:
graphql resolver will never return non-200 status

Reply:
https://github.com/toeverything/AFFiNE/actions/runs/13266531542/job/37035372428

I saw a 400 status response in last night test

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/hTwOityLamd4hitrae7M/6c8cae93-faaa-4848-9c8b-8610c59c03e6.png)

Reply:
strange

Reply:
![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/5c0ymolP9B7QStCsS1RP/643f1368-6c09-43b5-ae15-70cfedf966be.png)

I directly throw an error in the resolver and the response code is not 400

Reply:
Yes, I couldn't reproduce it last night either, so I changed the code a bit to see what the response content is. I'll modify it again to first check if the status is not 200, then expose the full body, so that next time we encounter it, we'll know what the issue is.

This 400 response is an unstable case.

Reply:
@forehalo https://app.graphite.dev/github/pr/toeverything/AFFiNE/10203/refactor-server-dont-convert-graphql-bad-request-into-internal-server-error

I reproduced a 400 request by creating an erroneous gql. Incidentally, I routinely filtered out server-side 500 errors for such illegal requests.